opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpanSource.java

```diff
@@ -36,7 +34,12 @@ public ThreadLocalActiveSpan activeSpan() {
 
     @Override
     public ActiveSpan makeActive(Span span) {
-        return new ThreadLocalActiveSpan(this, span, new AtomicInteger(1));
+        return new ThreadLocalActiveSpan(this, span, new AutoFinisher());
```

Comment:
I think there's a difference between updating a version and using the active span api. There's also a difference between writing code based on discussions like these and production experience with them. I wouldn't jump to conclusions that everything is fine.

What I don't like is that the liability of auto-finishing and its implementation are not only still here, but also still the default. Any problems with this implementation (as noted laboriously in the prior issue and I won't repeat here) are not less likely now!